### PR TITLE
AP-42 Send block reward only after the block is confirmed

### PR DESCRIPTION
**Pull request by georgebejan**

### CODE REVIEW

File: libs/paymentProcessor.js [hunk 1]
```diff
@@ -369,29 +378,34 @@ function SetupForPool(logger, poolOptions, setupFinished) {
...
-                            if (result.error && result.error.code === -6) {
-                                let higherPercent = withholdPercent + 0.01;
-                                logger.warning(logSystem, logComponent, 'Not enough funds to cover the tx fees for sending out payments, decreasing rewards by '
-                                    + (higherPercent * 100) + '% and retrying');
-                                trySend(higherPercent);
-                            }
-                            else if (result.error) {
-                                logger.error(logSystem, logComponent, 'Error trying to send payments with RPC sendmany '
-                                    + JSON.stringify(result.error));
+                        unlockAccountIfNecessary(poolAddress, poolOptions.addressPassword, function (isUnlocked) {
```
Comment (alinturbut) [inline libs/paymentProcessor.js:381]: I wouldn't try to unlock each time we do a payment, rather do this call at the intialization of the payment processor, maybe in a different file to extract a bit the logic.

File: pool_configs/aion.json [hunk 2]
```diff
@@ -2,10 +2,11 @@
     "enabled": true,
     "testnet": true,
     "coin": "aion.json",
-    "logLevel": "debug",
-    "__address_aion":    "0x604076f02ab252f3927e6a8721e2a18526e08182fd7b476a8f054765a60803b6",
+
+    "__address_aion":    "6a156bA44e87E1d621C266687eA6B63B2283cA5c",
```
Comment (synapsics-sergiu) [inline pool_configs/aion.json]: what is this address used for? should we have placeholders instead?

File: libs/paymentProcessor.js (hunk 1 above)
Comment (alinturbut) [reply libs/paymentProcessor.js:381]: @georgebejan, i know we discussed that we don't know about how long the account is locked. One idea would be to have this into the configuration, unlock it at initialisation of the payment, set it unlocked for X amount of time, then use setTimeout to unlock it every X amount of time.

File: pool_configs/aion.json (hunk 2 above)
Comment (georgebejan) [reply pool_configs/aion.json]: Yup, you're right. I'll change it

File: libs/paymentProcessor.js (hunk 1 above)
Comment (georgebejan) [reply libs/paymentProcessor.js:381]: 1. The payment processor will not reach this phase unless the pool has mined a confirmed pool, which will not be often enough that will cause a performance issue unlocking the account

2. If we unlock the account every X amount of time, we assume that the pool is the only part that interacts with our account. Maybe there are other parts (wallet, kernel) that can lock the account meanwhile, so I would say some sort of checking is needed each time

3. To reduce the number of checks, we could first try to send the transaction, and if we get an error, we unlock the account and then try to send the transaction again. This way, we make sure that we unlock the account only when it's necessary without any other check beforehand

Comment (alinturbut) [reply libs/paymentProcessor.js:381]: You are right. We can skip the performance part here until we get the sendmany

**Review by alinturbut — APPROVED** [review]

**Pull request merged**